theme/helpers.js

```diff
@@ -9,13 +13,94 @@ exports.Color = class {
         Object.keys(shades).forEach((shade) => (this[shade] = shades[shade]));
     }
 
-    shade(shade) {
+    alter(shade = 500) {
 
-        return this[shade];
+        return Color(this[shade]);
     }
 
     toString() {
 
         return this['500'];
     }
 };
+
+/**
+ * Converts an Eva Design theme into a MUI-style palette
+ * @returns {Object} Returns an object with color keys e.g. { primary: { 100: '#ffffff' }, secondary: { ... } }
+ */
+exports.paletteFromTheme = (theme) => {
+
+    const colors = Object.keys(theme).reduce((acc, key) => {
+
+        const regex = /^color-(?<color>\w+)-(?<transparent>\w+-)?(?<shade>\d{3})/;
+
+        Assert(regex.test(key), `Expected ${key} to match regex ${regex}`);
+
+        const { color, transparent, shade } = key.match(/^color-(?<color>\w+)-(?<transparent>\w+-)?(?<shade>\d{3})/).groups;
+
+        if (!(color in acc)) {
+            acc[color] = { transparent: {} };
+        }
+
+        if (transparent) {
+            acc[color].transparent[shade] = theme[key];
+        }
+        else {
+            acc[color][shade] = theme[key];
+        }
+
+        return acc;
+    }, {});
+
+    return Object.keys(colors).reduce((acc, color) => {
+
+        acc[color] = new exports.Color(colors[color]);
+        return acc;
+    }, {});
+};
+
+/**
+ * Provides a spacing value based on a scaling factor
+ * @example
+ * // returns 16
+ * Helpers.spacing(2)
+ * @example
+ * // returns 'auto 16px'
+ * Helpers.spacing('auto', 2)
+ * @returns {(number|string)} Returns an object with color keys e.g. { primary: { 100: '#ffffff' }, secondary: { ... } }
+ */
+exports.spacing = (...args) => {
+
+    if (args.length === 1) {
+        Assert(!isNaN(args[0]), `Expected '${args[0]}' to be a number`);
+        return parseInt(args[0], 10) * BASE_SCALING_FACTOR;
```

Comment:
Is there a reason we want to force these to `int` rather than `float`? Using this code in another project, I've occasionally needed to do little silly things like `1.5 * Theme.spacing(1)` to get a `12px` margin to match a design.